testsuites/listener/shared/client_sg/test_replication.py

```diff
@@ -227,6 +329,100 @@ def test_initial_push_replication(setup_client_syncgateway_test, continuous):
+    """
+    1. Prepare LiteServ to have 10000 documents.
+    2. Start continous push replication to push the docs into a sync_gateway database.
+    3. While docs are getting replcate , push the app to the background
+    4. Verify if all of the docs get pushed and replication continous when app goes background
+    """
+
+    sg_db = "db"
+    ls_db = "ls_db"
+    seth_channels = ["ABC", "NBC"]
+
+    cluster_config = setup_client_syncgateway_test["cluster_config"]
+    sg_mode = setup_client_syncgateway_test["sg_mode"]
+    ls_url = setup_client_syncgateway_test["ls_url"]
+    sg_one_admin = setup_client_syncgateway_test["sg_admin_url"]
+    sg_one_public = setup_client_syncgateway_test["sg_url"]
+    liteserv = setup_client_syncgateway_test["liteserv"]
+    liteserv_platform = setup_client_syncgateway_test["liteserv_platform"]
+    device_enabled = setup_client_syncgateway_test["device_enabled"]
+
+    sg_config = sync_gateway_config_path_for_mode("listener_tests/listener_tests", sg_mode)
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # No command to push the app to background on device, so avoid test to run on ios device
+    if((liteserv_platform.lower != "ios" and liteserv_platform.lower != "android") or (liteserv_platform.lower == "ios" and device_enabled)):
+        pytest.skip('This test only valid for mobile')
+    log_info("ls_url: {}".format(ls_url))
+    log_info("sg_one_admin: {}".format(sg_one_admin))
+    log_info("sg_one_public: {}".format(sg_one_public))
+
+    client = MobileRestClient()
+    client.create_user(sg_one_admin, sg_db, "seth", password="password", channels=seth_channels)
+    session = client.create_session(sg_one_admin, sg_db, "seth")
+
+    client.create_database(url=ls_url, name=ls_db)
+    bulk_docs_resp = []
+    if need_attachments:
+        doc_bodies = document.create_docs(
+            doc_id_prefix="seeded_doc",
+            number=num_docs,
+            attachments_generator=attachment.generate_2_png_100_100,
+            channels=seth_channels
+        )
+    else:
+        doc_bodies = document.create_docs(doc_id_prefix='seeded_doc', number=num_docs, channels=seth_channels)
+
+    for x in xrange(0, len(doc_bodies), 100000):
+        chunk_docs = doc_bodies[x:x + 100000]
+        ch_bulk_docs_resp = client.add_bulk_docs(url=ls_url, db=ls_db, docs=chunk_docs, auth=session)
+        log_info("length of bulk docs resp{}".format(len(ch_bulk_docs_resp)))
+        bulk_docs_resp += ch_bulk_docs_resp
+    assert len(bulk_docs_resp) == num_docs
+
+    # Start push replication with app background
+    if replication_after_backgroundApp:
+        liteserv.close_app()
+        time.sleep(2)
+        client.start_replication(
+            url=ls_url,
+            continuous=True,
+            from_db=ls_db,
+            to_url=sg_one_admin,
+            to_db=sg_db
+        )
+    else:
+        client.start_replication(
```

Comment:
Move client.start_replication() out of the if else

Reply:
same comment as above . I made to one line. I know I felt same as you when I see the code first time, then I realize

Reply:
Yes, its a single call, but is written twice. I meant something like this:

if replication_after_backgroundApp:
liteserv.close_app()

client.start_replication()

if not replication_after_backgroundApp:
liteserv.close_app()

Reply:
same as previous comment, changed according to your request and added comment .